src/editors/playgroundController.ts

```diff
@@ -315,15 +315,24 @@ export default class PlaygroundController {
     );
   }
 
-  private openResultAsVirtualDocument(viewColumn) {
-    vscode.workspace
-      .openTextDocument(
-        this.getVirtualDocumentUri(this.playgroundResult?.content)
-      )
-      .then((doc) => {
+  private openResultAsVirtualDocument(viewColumn: vscode.ViewColumn) {
+    const content =
+      this.playgroundResult && this.playgroundResult.content
+        ? this.playgroundResult.content
+        : '';
+
+    vscode.workspace.openTextDocument(this.getVirtualDocumentUri(content)).then(
+      (doc) => {
         this._playgroundResultTextDocument = doc;
         vscode.window.showTextDocument(doc, { preview: false, viewColumn });
-      });
+      },
+      (error) => {
+        vscode.window.showErrorMessage(
+          `Unable to open results document: ${error.message}`
+        );
+        log.error('Open playground ERROR', error);
+      }
+    );
```

Comment:
Missing error catching here was causing the same error in tests, that was described in this ticket:
https://jira.mongodb.org/browse/VSCODE-204

Let's further keep an eye on it.

Reply:
We previously had an issue with catching unhandled promise rejections for `vscode.workspace.openTextDocument` and only catching errors as a second argument solved the issue: `.then(() => { /* success */ }, () => { /* fail */ });`. You can see that we do it all the time here in playgroundController.ts.

But your point about awaiting `this.openResultAsVirtualDocument(viewColumn)` results is totally fair. The only thing that instead of returning `return this.openResultAsVirtualDocument(viewColumn)` tenable result I will just await result and return `return resolve(false);` because the expected result of the function is boolean.